### PR TITLE
OCT-1725: Combine dozens of frontend requests into just a few

### PR DESCRIPTION
**Pull request by jmikolajczyk**
## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: client/src/hooks/queries/donors/useProjectsDonors.ts [hunk 1]
```diff
@@ -17,57 +11,47 @@ export default function useProjectsDonors(epoch?: number): {
...
-        (isDecisionWindowOpen === true || epoch !== undefined),
-      queryFn: () => apiGetProjectDonors(projectAddress, epoch || currentEpoch! - 1),
-      queryKey: QUERY_KEYS.projectDonors(
-        projectAddress,
-        epoch ?? (isDecisionWindowOpen ? currentEpoch! - 1 : currentEpoch!),
-      ),
-      select: response => mapDataToProjectDonors(response),
-    })),
-  });
+  // TODO OCT-1139 implement socket here.
```
Comment (aziolek) [inline client/src/hooks/queries/donors/useProjectsDonors.ts:18]: Socket got cut off, but comment stayed?

File: client/src/components/Metrics/MetricsEpoch/MetricsEpochGridBelowThreshold/MetricsEpochGridBelowThreshold.tsx [hunk 2]
```diff
@@ -34,12 +35,9 @@ const MetricsEpochGridBelowThreshold: FC<MetricsEpochGridBelowThresholdProps> =
     },
   }));
   const { data: cryptoValues, error } = useCryptoValues(displayCurrency);
-  const { data: projectsDonors } = useProjectsDonors(
-    isDecisionWindowOpen && epoch === lastEpoch ? undefined : epoch,
-  );
 
   const projectsBelowThreshold =
-    Object.keys(projectsDonors).length -
+    (projectsEpoch?.projectsAddresses.length || 0) -
```
Comment (aziolek) [inline client/src/components/Metrics/MetricsEpoch/MetricsEpochGridBelowThreshold/MetricsEpochGridBelowThreshold.tsx:40]: What is the reason for this change?

**Review by aziolek — CHANGES_REQUESTED** [review]
So now instead of fetching donors per project we need to fetch all the allocations in the epoch and reduce them ourselves? Seems like a step backwards.

File: client/src/components/Metrics/MetricsEpoch/MetricsEpochGridCurrentDonors/MetricsEpochGridCurrentDonors.tsx [hunk 3]
```diff
@@ -16,7 +17,7 @@ const MetricsEpochGridCurrentDonors: FC<MetricsEpochGridCurrentDonorsProps> = ({
   const { epoch } = useMetricsEpoch();
   const { data: epochAllocations } = useEpochAllocations(epoch);
 
-  const currentDonorsString = `${epochAllocations?.length || 0}`;
+  const currentDonorsString = uniqBy(epochAllocations, 'donor').length.toString();
```
[the remark this answers is not in the document]
Comment (jmikolajczyk) [reply client/src/components/Metrics/MetricsEpoch/MetricsEpochGridCurrentDonors/MetricsEpochGridCurrentDonors.tsx:20]: We need to count donors and one donor can make more than 1 allocation.

File: client/src/components/Metrics/MetricsEpoch/MetricsEpochGridBelowThreshold/MetricsEpochGridBelowThreshold.tsx (hunk 2 above)
Comment (jmikolajczyk) [reply client/src/components/Metrics/MetricsEpoch/MetricsEpochGridBelowThreshold/MetricsEpochGridBelowThreshold.tsx:40]: `projectsDonors` return only projects that have received donation (here we need all projects).

File: client/src/hooks/queries/donors/useProjectsDonors.ts (hunk 1 above)
Comment (jmikolajczyk) [reply client/src/hooks/queries/donors/useProjectsDonors.ts]: Comment removed

**Comment by jmikolajczyk** [comment]
> So now instead of fetching donors per project we need to fetch all the allocations in the epoch and reduce them ourselves? Seems like a step backwards.

I don't think so. Instead of making request for each project donors we do only one and reduce data on our side.

File: client/src/hooks/queries/donors/useProjectsDonors.ts (hunk 1 above)
Comment (aziolek) [reply client/src/hooks/queries/donors/useProjectsDonors.ts]: Why socket got cut off, then?

**Review by aziolek — APPROVED** [review]
